Integration with data pipeline api for parameters

vinopm opened the pull request.

Commit:
Integration with data pipeline api for parameters

vinopm commented:
@sam-brett @johnnonweiler this works AFAIK, we may have to change the config.yaml/metadata.yaml file a bit when we try the integrated run today. Will update the PR if we end up changing the configuration, but the rest is ready to be reviewed

Commit:
Clean up

Commit:
Fix tests

Commit:
Use namespace instead of data product directly + upload new metadata.yaml

Commit:
Update suggestion from John for cross platform compatibility

Commit:
Update suggestion from John for cross platform compatibility

Commit:
change std api version number for dependency

Commit:
Revert change of data_directory. Data pipeline team have fixed download script error

Commit:
Merge branch 'develop' into vinopm-integrate-with-data-pipeline-api

johnnonweiler committed:
Merge branch 'develop' into vinopm-integrate-with-data-pipeline-api

There were many conflicts which had to be manually resolved, and the new
ageDependence parameter was (for now) removed, because it was too
complicated to merge that change - I think changes will be needed to the
data pipeline api code before that can be done.

# Conflicts:
#	input/populationSettings.json
#	src/main/java/uk/co/ramp/AppConfig.java
#	src/main/java/uk/co/ramp/event/AlertEventProcessor.java
#	src/main/java/uk/co/ramp/event/CommonVirusEventProcessor.java
#	src/main/java/uk/co/ramp/event/EventContext.java
#	src/main/java/uk/co/ramp/io/readers/PopulationPropertiesReader.java
#	src/test/java/uk/co/ramp/event/AlertEventProcessorTest.java
#	src/test/java/uk/co/ramp/io/readers/PopulationPropertiesReaderTest.java
#	src/test/java/uk/co/ramp/people/PopulationGeneratorTest.java

johnnonweiler commented:
@sam-brett - Merging this with the latest develop was hard.  It was so tricky that I gave up trying to merge everything, and removed the new ageDependence feature.  There were _many_ manual changes required to fix merge conflicts.  Please could you check the merge commit?

In particular, please check AlertEventProcessor.java - I changed the behaviour of timeInStatusAndTestQueue() where I was confident the old behaviour was wrong. I suspect that other bits may also have been incorrect - returning 1 day where they should be returning 1 time step.  Also, it is likely that I have made mistakes in here!

I have changed one of the toml files, so the files on this branch no longer correspond to the files which were uploaded to the data pipeline.

The new infectionRates.json parameters should probably come from the data pipeline, but I have not yet made that change.

johnnonweiler committed:
Set .toml files to binary in .gitattributes

I was having problems on Windows with automatic conversion of line endings
changing files so that hash verification then failed.

sam-brett commented on src/main/java/uk/co/ramp/policy/isolation/BoundedDistributionSerializer.java:
```diff
@@ -0,0 +1,65 @@
+    try {
+      gen.writeObjectField(entry.getKey(), entry.getValue());
+    } catch (IOException e) {
+      e.printStackTrace();
```
This exception should be handled

sam-brett replied:
This exception should be handled

sam-brett commented on build.gradle:
```diff
@@ -30,7 +31,9 @@ dependencies {
     implementation 'com.fasterxml.jackson.dataformat:jackson-dataformat-csv:2.11.0'
     implementation "org.immutables:value-annotations:2.8.8"
     implementation "org.immutables:gson:2.8.8"
```
I guess this is tech debt that I can deal with, but GSON can be removed in favour of Jackson now. GSON is only used in tests now.

sam-brett commented on src/main/java/uk/co/ramp/policy/isolation/BoundedDistributionSerializer.java:
```diff
@@ -0,0 +1,65 @@
+ * can/will be removed once we migrate ALL 'distribution' related objects write to the data
+ * pipeline.
+ */
+public class BoundedDistributionSerializer extends JsonSerializer<BoundedDistribution> {
```
This class isn't tested

sam-brett reviewed (approved):
@johnnonweiler Happy to approve and I'll fix the age dependency and other bits in another PR.
